homeassistant/components/rest/sensor.py

```diff
@@ -99,35 +130,50 @@ def __init__(
         coordinator: DataUpdateCoordinator[None] | None,
         rest: RestData,
         config: ConfigType,
-        unique_id: str | None,
+        trigger_entity_config: ConfigType,
+        unit_of_measurement: str | None,
+        state_class: str | None,
     ) -> None:
         """Initialize the REST sensor."""
+        ManualTriggerEntity.__init__(self, hass, trigger_entity_config)
         RestEntity.__init__(
             self,
             coordinator,
             rest,
             config.get(CONF_RESOURCE_TEMPLATE),
             config[CONF_FORCE_UPDATE],
         )
-        TemplateSensor.__init__(
-            self,
-            hass,
-            config=config,
-            fallback_name=DEFAULT_SENSOR_NAME,
-            unique_id=unique_id,
-        )
+        self._attr_native_unit_of_measurement = unit_of_measurement
+        self._attr_state_class = state_class
```

Comment:
Maybe add a `ManualTriggerSensor` which knows about uom amd state class, similar to `TemplateSensor`?